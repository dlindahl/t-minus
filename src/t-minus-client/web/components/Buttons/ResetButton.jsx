import { bindActionCreators } from 'redux'
import AppNavButton from './AppNavButton'
import { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import AutoRenew from 'react-icons/lib/md/autorenew'
import { resetClock } from '../../../shared/actions/ClockActions'

function getActions (dispatch) {
  return bindActionCreators({ resetClock }, dispatch)
}

@connect(null, getActions)
export default class ResetButton extends Component {
  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
  }
  handleClick () {
    this.props.resetClock()
  }
  render () {
    return (
      <AppNavButton label="Reset" onClick={this.handleClick}>
        <AutoRenew/>
      </AppNavButton>
    )
  }
}

ResetButton.propTypes = {
  resetClock: PropTypes.func
}
