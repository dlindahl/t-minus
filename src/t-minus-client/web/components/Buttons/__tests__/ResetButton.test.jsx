import configureStore from 'redux-mock-store'
import emptyObj from 'empty/object'
import renderer from 'react-test-renderer'
import ResetButton from '../ResetButton'
import { Provider } from 'react-redux'

const STATE = {
  display: emptyObj
}
const store = configureStore()

describe('ResetButton', () => {
  it('renders the component', () => {
    const component = renderer.create(
      <Provider store={store(STATE)}>
        <ResetButton/>
      </Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('resets the clock when reset', () => {
    const resetClock = jest.fn()
    const component = renderer.create(
      <Provider store={store(STATE)}>
        <ResetButton.WrappedComponent
          resetClock={resetClock}
        />
      </Provider>
    )
    const tree = component.toJSON()
    tree.props.onClick()
    expect(resetClock).toHaveBeenCalled()
  })
})
