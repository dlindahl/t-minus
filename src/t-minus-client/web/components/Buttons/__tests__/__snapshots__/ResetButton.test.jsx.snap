exports[`ResetButton renders the component 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "background": "transparent",
      "border": 0,
      "color": "var(--Silver)",
      "cursor": "pointer",
      "fontSize": "2rem",
      "height": 50,
      "lineHeight": 1,
      "minWidth": 50,
      "padding": 0,
    }
  }>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "top",
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m31.3 12.9c1.3 2 2.1 4.5 2.1 7.1 0 7.3-6.1 13.4-13.4 13.4v5l-6.6-6.8 6.6-6.6v5c5.5 0 10-4.4 10-10 0-1.7-0.5-3.3-1.2-4.7z m-11.3-2.9c-5.5 0-10 4.5-10 10 0 1.7 0.4 3.3 1.2 4.7l-2.4 2.4c-1.4-2-2.2-4.5-2.2-7.1 0-7.3 6.1-13.4 13.4-13.4v-5l6.6 6.8-6.6 6.6v-5z" />
    </g>
  </svg>
  <br />
  <div
    style={
      Object {
        "color": "var(--Silver)",
        "fontFamily": "Roboto Mono, monospace",
        "fontSize": ".8rem",
        "lineHeight": 1.5,
        "textTransform": "uppercase",
      }
    }>
    Reset
  </div>
</button>
`;
